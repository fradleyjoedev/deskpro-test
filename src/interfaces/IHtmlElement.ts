export interface IHTMLElement {
  componentType: string;
  title: string;
  description: string;
}