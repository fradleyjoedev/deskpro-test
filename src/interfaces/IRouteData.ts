export interface IRouteData {
  path: string;
  component: string;
  linkTitle: string;
  pageTitle: string;
  pageDescription: string;
}