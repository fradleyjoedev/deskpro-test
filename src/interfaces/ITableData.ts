export interface ITableData {
  heading: string;
  data: string[];
};