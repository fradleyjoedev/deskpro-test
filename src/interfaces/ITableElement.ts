export interface ITableElement {
  componentType: string;
}