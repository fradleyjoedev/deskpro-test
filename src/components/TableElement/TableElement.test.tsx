import React from "react";
import { shallow } from 'enzyme';
import { TableElement } from './index';
import { ITableElement } from '../../interfaces/ITableElement';

const data: ITableElement = {
  componentType: 'HTMLElement',
  columns: [
    { "heading": "Name", "data": ["Billy", "Joe", "Dave"]}, 
    { "heading": "Email", "data": ["billy@mcgun.co.uk", "joe@hotmail.com", "dave@gtimes.com"]},
    { "heading": "Location", "data": ["Manchester", "Newcastle", "London"]}
  ]
}

describe('Test for TableElement', () => {
  it('snapshot', () => {
    const link = shallow(<TableElement {...data} />);
    expect(link).toMatchSnapshot();
  });
});