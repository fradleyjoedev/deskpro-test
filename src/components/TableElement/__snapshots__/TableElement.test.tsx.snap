// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for TableElement snapshot 1`] = `
<div
  className="table spacing"
>
  <ul
    className="table__column"
    key="0"
  >
    <li
      className="table__column--item"
    >
      Name
    </li>
    <li
      className="table__column--item"
      key="0"
    >
      Billy
    </li>
    <li
      className="table__column--item"
      key="1"
    >
      Joe
    </li>
    <li
      className="table__column--item"
      key="2"
    >
      Dave
    </li>
  </ul>
  <ul
    className="table__column"
    key="1"
  >
    <li
      className="table__column--item"
    >
      Email
    </li>
    <li
      className="table__column--item"
      key="0"
    >
      billy@mcgun.co.uk
    </li>
    <li
      className="table__column--item"
      key="1"
    >
      joe@hotmail.com
    </li>
    <li
      className="table__column--item"
      key="2"
    >
      dave@gtimes.com
    </li>
  </ul>
  <ul
    className="table__column"
    key="2"
  >
    <li
      className="table__column--item"
    >
      Location
    </li>
    <li
      className="table__column--item"
      key="0"
    >
      Manchester
    </li>
    <li
      className="table__column--item"
      key="1"
    >
      Newcastle
    </li>
    <li
      className="table__column--item"
      key="2"
    >
      London
    </li>
  </ul>
</div>
`;
